packages/terra-time-input/src/TimeUtil.js

```diff
@@ -1,11 +1,143 @@
+   */
+  static incrementHour(hour, timeVariant) {
+    const maxValue = timeVariant === this.FORMAT_12_HOUR ? 12 : 23;
+
+    // Increment the value by 1 when arrow up is pressed.
+    if (hour) {
+      let numericMinute = Number(hour);
+
+      if (numericMinute < maxValue) {
+        numericMinute += 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      } else if (timeVariant === this.FORMAT_12_HOUR) {
+        return '01';
+      }
+
+      return hour;
+    }
+
+    return timeVariant === this.FORMAT_12_HOUR ? '12' : '00';
+  }
+
+  /**
+   * Decrements an hour to it's next value. Resets to 12 if the passed in hour is 01 and the variant is a 12 hour clock.
+   * @param {String} hour Hour to decrement
+   * @param {String} timeVariant Variant of time the passed in hour refers to.
+   * @return {String} Returns a string representation of the value of the an hour after it's decremented. If the
+   *   hour is 12 and the timeVariant is 12, it returns 12.
+   */
+  static decrementHour(hour, timeVariant) {
+    const minValue = timeVariant === this.FORMAT_12_HOUR ? 1 : 0;
+
+    if (hour) {
+      let numericMinute = Number(hour);
+
+      if (numericMinute > minValue) {
+        numericMinute -= 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      } else if (timeVariant === this.FORMAT_12_HOUR) {
+        return '12';
+      }
+
+      return hour;
+    }
+
+    return timeVariant === this.FORMAT_12_HOUR ? '12' : '00';
+  }
+
+  /**
+   * Increments a minute to it's next value
+   * @param {String} minute Minute to increment
+   * @return {String} Returns a string representation of the value of the passed in minute after it's incremented
+   */
+  static incrementMinute(minute) {
+    if (minute) {
+      let numericMinute = Number(minute);
+
+      if (numericMinute < 59) {
+        numericMinute += 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      }
+
+      return minute;
+    }
+
+    return '00';
+  }
+
+  /**
+   * Decrements a minute to it's next value
+   * @param {String} minute Minute to decrement
+   * @return {String} Returns a string representation of the value of the passed in minute after it's decremented
+   */
+  static decrementMinute(minute) {
+    if (minute) {
+      let numericMinute = Number(minute);
+
+      if (numericMinute > 0) {
+        numericMinute -= 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      }
+
+      return minute;
+    }
+
+    return '00';
+  }
+
   static splitHour(time) {
     if (typeof (time) === 'string') {
       const hourAndMinute = time.split(':');
 
       if (hourAndMinute.length) {
         const hour = Number(hourAndMinute[0]);
         if (hour >= 0 && hour < 24) {
+          if (hourAndMinute[0].length === 1) {
```

Comment:
Should this be moved before ` const hour = Number(hourAndMinute[0]);`?

Reply:
The one thing I like about it being here is that the string is only concatenated if the hour is between 0 and 23. If we move it up, we could be doing concatenation when it's not necessary.